src/python/WMCore/WorkQueue/WorkQueue.py

```diff
@@ -818,95 +858,57 @@ def pullWork(self, resources=None):
-            for element in workflowsToCheck:
-                # Easy check, close elements with no defined OpenRunningTimeout
-                policy = element.get('StartPolicy', {})
-                openRunningTimeout = policy.get('OpenRunningTimeout', 0)
-                if not openRunningTimeout:
-                    # Closing, no valid OpenRunningTimeout available
-                    workflowsToClose.append(element.id)
-                    continue
-
-                # Check if new data is currently available
-                skipElement = False
-                spec = self.backend.getWMSpec(element.id)
-                for topLevelTask in spec.taskIterator():
-                    policyName = spec.startPolicy()
-                    if not policyName:
-                        raise RuntimeError("WMSpec doesn't define policyName, current value: '%s'" % policyName)
-
-                    policyInstance = startPolicy(policyName, self.params['SplittingMapping'],
-                                                 rucioObj=rucioObj, logger=self.logger)
-                    if not policyInstance.supportsWorkAddition():
-                        continue
-                    if policyInstance.newDataAvailable(topLevelTask, element):
-                        skipElement = True
-                        self.backend.updateInboxElements(element.id, TimestampFoundNewData=currentTime)
-                        msg = "There are blocks still open for writing in DBS."
-                        self.logdb.post(element['RequestName'], msg, "warning")
-                        break
-                if skipElement:
-                    continue
-
-                # Check if the delay has passed
-                newDataFoundTime = element.get('TimestampFoundNewData', 0)
-                childrenElements = self.backend.getElementsForParent(element)
-                if childrenElements:
-                    lastUpdate = float(max(childrenElements, key=lambda x: x.timestamp).timestamp)
-                    if (currentTime - max(newDataFoundTime, lastUpdate)) > openRunningTimeout:
-                        workflowsToClose.append(element.id)
-                    # if it is successful remove previous error
-                    self.logdb.delete(element.id, "error", this_thread=True)
-                else:
-                    msg = "ChildElement is empty for element id %s: investigate" % element.id
-                    self.logdb.post(element.id, msg, "error")
-                    # self.logdb.upload2central(element.id)
-                    self.logger.error(msg)
+        workflowsToCheck = self.backend.getInboxElements(OpenForNewData=True)
+        self.logger.info("Retrieved a list of %d open workflows", len(workflowsToCheck))
+        workflowsToClose = []
+        currentTime = time.time()
+        for element in workflowsToCheck:
+            # fetch attributes from the inbox workqueue element
+            openRunningTimeout = element.get('OpenRunningTimeout', 0)
```

Comment:
This last minute change actually broke this functionality. The WorkQueue element data structure does not have this parameter... 
I will get it fixed in: https://github.com/dmwm/WMCore/pull/11135